### DISCUSSION TITLE
Undefined in SpecificFriendCategoryItemView.JS / useParams-- Needs UID and categoryKey

### DISCUSSION THREAD

**Discussion by thedigitalmenagerie** [nss-evening-cohort-14/evening-client-side#318] [Capstones!] [answered]
Honey-Rae Swan

I am trying to get category items, specific to friends, and I my url is returning undefined. I know I need to get them based on the chosen user's UID and the specific categories' firebasekey (which translates to item categoryKeys).

List any error messages that appear
vendors~main.chunk.js:5410 A boolean is being passed as a fourth parameter to window.open. This is not used and may cause an exception in a future release.

What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)
I honestly don't even know what to google. 

What have you tried? 
I am pulling up this view from a card and tried passing the info from the card to the other view. I got undefined this way. 
I tried making new calls and making new calls to the API. That is what I've pushed up. 

Any API Responses you are getting.
No external API's.

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with NO SCREEN SHOTS 🥰
https://github.com/thedigitalmenagerie/hr-gratitude-capstone/blob/friendViews/src/Components/FriendComponents/SpecificFriendCategoryCardComponent.js
https://github.com/thedigitalmenagerie/hr-gratitude-capstone/blob/friendViews/src/Views/FriendViews/SpecificFriendSingleCategoryView.js
https://github.com/thedigitalmenagerie/hr-gratitude-capstone/blob/friendViews/src/Components/FriendComponents/SpecificFriendCategoryItemCardComponent.js

**Comment by TrinityChristiana** [comment]
@thedigitalmenagerie Will add their solution!

**Comment by thedigitalmenagerie** [comment, marked as answer]
import { useHistory, useLocation } from 'react-router-dom';  

const location = useLocation();

  const handleClick = (type) => {
    switch (type) {
      case 'view':
        console.warn(location);
        console.warn(history);
        // history.push(`/myCategories/${firebaseKey}`);
        // history.push(`/friendView/${firebaseKey}/Categories/${uid}/CategoryItems/${firebaseKey}`);
        history.push(`${location.pathname}/CategoryItems/${firebaseKey}`);
        break;
      default:
        console.warn('Nothing selected');
    }
  };

When pushing to category specific items in the logged in user's view, I was using the firebaseKey as opposed to categoryKey in the history push. Here, I was using categoryKey. This is why it was working with the same call I used in the logged in user's category items. I couldn't use this because I needed the pathway to be specific to the selected friend. 

This presented a new issue, as this pathway could not differentiate between the friend's firebaseKey and the category's firebaseKey. 

In order to use one firebaseKey (specific to category) and remove the reference to the friends' firebaseKey, Trinity pointed me to 
https://stackoverflow.com/questions/42253277/react-router-v4-how-to-get-current-route

location.pathname solved the two issues above!
